### PR TITLE
Search travel

### PR DESCRIPTION
**Pull request by wenyuanli0326**
1. Log in as Anthony Gray.
2. Edit the travel information and enter a city, for example, Seattle. Set the start date to be within the next 30 days.
3. Navigate to the search page, enter the city in the search field, and click on the search button.
4. Verify that Anthony Gray appears in the search results.
5. Attempt to set the start date to be outside of the next 30 days.
6. Perform another search and verify that Anthony Gray no longer appears in the search results.

I also fix the country/state selection problem and set the city to be an input box. 

For tests:
run cucumber features/search_travel.feature 

Tests results:
search_travel_test

### CODE REVIEW

File: app/models/general_info.rb [hunk 1]
```diff
@@ -338,7 +338,23 @@ def self.filterBy state, profession, city
     #  puts "users are: #{room[:first_name]}"
     #end
 
-    return @filteredUsers
+    puts "check adfsaf"
```
Comment (AJ1904) [inline app/models/general_info.rb:341]: required?

File: app/models/general_info.rb [hunk 2]
```diff
@@ -338,7 +338,23 @@ def self.filterBy state, profession, city
...
+    puts city.present?
```
Comment (AJ1904) [inline app/models/general_info.rb:342]: required?

File: app/models/general_info.rb [hunk 3]
```diff
@@ -338,7 +338,23 @@ def self.filterBy state, profession, city
     #  puts "users are: #{room[:first_name]}"
     #end
 
-    return @filteredUsers
+    puts "check adfsaf"
+    puts city.present? 
+    puts city
```
Comment (AJ1904) [inline app/models/general_info.rb:343]: required?

**Review by AJ1904 — CHANGES_REQUESTED** [review]
1. Update the travel statement according to what client asked for.
In city,state from 3-letter-month day to 3-letter-month day : [description]
2. Put validation - start date cannot be more than end date
3. Put validation -  current - 30 days <= start date <= current + 30 days
4. Travel info should be removed from database once current > end date

**Review by AJ1904 — CHANGES_REQUESTED** [review]
The app links (fb,linkedin,etc) should not go out of the box. They are inside the box in heroku, please fix in this branch.
image

**Comment by wenyuanli0326** [comment]
Please check the commit history. Everything you mentioned are fixed.

**Comment by AJ1904** [comment]
Test case fails (because of the recent changes to general_steps.rb). Please update test case accordingly.
image

**Review by baichuan0113 — APPROVED** [review]

**Review by AJ1904 — APPROVED** [review]

**Comment by wenyuanli0326** [comment]
Modified test due to changes in general_steps. 

Test pass:
search_travel_test1

**Pull request merged**